src/metabase/api/collection.clj

```diff
@@ -59,14 +59,16 @@
 
 (defn- remove-other-users-personal-subcollections
   [user-id collections]
-  (let [personal-ids        (t2/select-fn-set :id :model/Collection
-                                              {:where
-                                               [:and [:!= :personal_owner_id nil] [:!= :personal_owner_id user-id]]})
+  (let [personal-ids         (t2/select-fn-set :id :model/Collection
+                                               {:where
+                                                [:and [:!= :personal_owner_id nil] [:!= :personal_owner_id user-id]]})
         personal-descendant? (comp personal-ids
```

Comment:
Another option could just be to call `set` on `personal-ids` to coerce `nil` into an empty set.

TBH — I'm not a huge fan of using `comp` to build up a function like this when it has more than 2-3 steps, since I think it's a little less readable than just using an anonymous function and maybe a `let` binding to describe what's happening. Something like
```
(fn [collection]
  (let [first-parent-collection-id ...]
    ((set personal-ids) first-parent-collection-id)))
```

Won't block the PR on this since it's a style thing but conciseness != readability

Reply:
Or even just a `->` macro within the anonymous function so at least you can read the steps top down instead of bottom up 🤷‍♂️ but again, a style thing